### PR TITLE
chore(website): esquery is not working at TypescriptNode

### LINKED ISSUE #8896: Bug: Website ESQuery doesn't work for typescript nodes
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

website

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.3&showAST=es&fileType=.tsx&esQuery=N4IgZglgNgLgpgJxALhAFQMpoJ4Ac4CCAdkQPYwCGMEpRIANCAM5xRwDGMpSyoMecFCAgATOEWqREDEADcKUAK6DUmHPmJlK1WiAC%2BeoA&code=FAYw9gdgzgLgBLATgLgTRBLCBzOBeOAcgBYAmQoA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const str: string = '42'
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
  }
}
```

### Expected Result

I expected to find the node when looking for it using the esquery query `TSTypeAnnotation`

### Actual Result

I get an empty list

### Additional Info

Probably we need to pass the visitorKeys as options to match (https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/website/src/components/ast/ASTViewer.tsx#L27)

### Versions

/

### PR DESCRIPTION
**Pull request by developer-bandi**
## PR Checklist

- [x] Addresses an existing open issue: fixes #8896
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

- add visitor key package in website package
- add option type in esquery type

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -19304,11 +19304,11 @@ __metadata:
 
 "typescript@patch:typescript@npm%3A5.4.3#~builtin<compat/typescript>":
   version: 5.4.3
-  resolution: "typescript@patch:typescript@npm%3A5.4.3#~builtin<compat/typescript>::version=5.4.3&hash=5adc0c"
+  resolution: "typescript@patch:typescript@npm%3A5.4.3#~builtin<compat/typescript>::version=5.4.3&hash=f3b441"
   bin:
     tsc: bin/tsc
     tsserver: bin/tsserver
-  checksum: 523ec29bc308b4ca0cec9a139ac08a27939891d8ad620d249cf5e1a35454e37cd7bb52613a1884f1206aa11d2fb1c4c83fb9e939bde542aaa8527ba818cb8177
+  checksum: 3a62fe90aa79d68c9ce38ea5edb2957e62801c733b99f0e5a2b8b50922761f68f7d9a40d28c544b449866e81185cddb93cba2496d0ff3fa52ef5b1f8bcace38c
```
Comment (armano2) [inline yarn.lock:19311]: build seem to fail due to checksum change, can you revert this?

Comment (developer-bandi) [reply yarn.lock:19311]: sure, thank you!

File: packages/website/src/components/ast/ASTViewer.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { visitorKeys } from '@typescript-eslint/visitor-keys';
```
Comment (armano2) [inline packages/website/src/components/ast/ASTViewer.tsx:1]: we should import it from `website-eslint` package as we do not want to include 2nd copy of this package in project

**Review by armano2 — APPROVED** [review]

**Review by JoshuaKGoldberg — APPROVED** [review]
Swell, thanks!
🚀

**Pull request merged**